lineman/app/models/group_model.coffee

```diff
@@ -148,6 +148,12 @@ angular.module('loomioApp').factory 'GroupModel', (DraftableModel, AppConfig) ->
         @remove()
         _.each @memberships(), (m) -> m.remove()
 
+    markAsRead: =>
+      @remote.postMember(@key, 'mark_as_read')
+      _.map @discussions(), (discussion) ->
+        return unless discussion.isUnread()
+        discussion.update(lastReadAt: moment(), readSalientItemsCount: discussion.salientItemsCount)
```

Comment:
What failures are possible?

I think the weird guess at values could be legitimate.. If we use the
values from the last known discussion item. Any newer data can just arrive
down the message channel. I can't see why it's a problem to update to
latest known values client side.

We really do want this to update immediately, user testing tells us that
it's a really bad time to click "go away now" and have to wait. Also
marking many discussions as read is really slow on the server side.

On Fri, 11 Dec 2015 at 15:53 James Kiesel <notifications@github.com> wrote:

> In lineman/app/models/group_model.coffee
> <https://github.com/loomio/loomio/pull/2796#discussion_r47317156>:
>
> > @@ -148,6 +148,12 @@ angular.module('loomioApp').factory 'GroupModel', (DraftableModel, AppConfig) ->
> >          @remove()
> >          _.each @memberships(), (m) -> m.remove()
> >
> > +    markAsRead: =>
> > +      @remote.postMember(@key, 'mark_as_read')
> > +      _.map @discussions(), (discussion) ->
> > +        return unless discussion.isUnread()
> > +        discussion.update(lastReadAt: moment(), readSalientItemsCount: discussion.salientItemsCount)
>
> Actually the more I look at this the more I don't like it. We're getting
> out of sync with the server here, don't have a good way of handling
> failure, and are left with weirdo guessed-at values designed to make
> isUnread() false, which don't get updated because we're not responding
> with anything useful from the server. We also don't update other attributes
> like readCommentsCount and readItemsCount here, meaning they're out of
> sync on the record itself.
>
> I think we need to just wait for the request to come back, perhaps marking
> the threads with something like readOnClient which could exclude them
> from the unread count in the meantime.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2796/files#r47317156>.
>